test/modules/ref2vec-centroid/centroid_test.go

```diff
@@ -0,0 +1,152 @@
+
+		res := helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.Nil(t, res.Vector)
+		assert.Equal(t, article.ID, res.ID)
+
+		article.WithReferences(
+			&models.SingleRef{Beacon: newBeacon(para1.Class, para1.ID)},
+			&models.SingleRef{Beacon: newBeacon(para2.Class, para2.ID)},
+		)
+
+		helper.UpdateObject(t, article.Object())
+
+		res = helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.Equal(t, article.ID, res.ID)
+		expectedVec := []float32{3, 5, 7}
+		assert.EqualValues(t, expectedVec, res.Vector)
+	})
+
+	t.Run("create object with references, remove references", func(t *testing.T) {
+		ref1 := &models.SingleRef{Beacon: newBeacon(para1.Class, para1.ID)}
+		ref2 := &models.SingleRef{Beacon: newBeacon(para2.Class, para2.ID)}
+
+		article := articles.NewArticle().
+			WithTitle("Popularity of the Nissan RB26DETT").
+			WithReferences(ref1, ref2)
+		helper.CreateObject(t, article.Object())
+		defer helper.DeleteObject(t, article.Object())
+
+		res := helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		expectedVec := []float32{3, 5, 7}
+		assert.EqualValues(t, expectedVec, res.Vector)
+
+		helper.DeleteReference(t, article.Object(), ref2, "hasParagraphs")
+		res = helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.EqualValues(t, para1.Vector, res.Vector)
+
+		helper.DeleteReference(t, article.Object(), ref1, "hasParagraphs")
+		res = helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.Nil(t, res.Vector)
+	})
+
+	t.Run("create object add references, remove references", func(t *testing.T) {
+		ref1 := &models.SingleRef{Beacon: newBeacon(para1.Class, para1.ID)}
+		ref2 := &models.SingleRef{Beacon: newBeacon(para2.Class, para2.ID)}
+
+		article := articles.NewArticle().
+			WithTitle("Popularity of the Nissan RB26DETT")
+		helper.CreateObject(t, article.Object())
+		defer helper.DeleteObject(t, article.Object())
+
+		res := helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.Nil(t, res.Vector)
+
+		helper.AddReference(t, article.Object(), ref1, "hasParagraphs")
+		res = helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.EqualValues(t, para1.Vector, res.Vector)
+
+		helper.AddReference(t, article.Object(), ref2, "hasParagraphs")
+		res = helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.EqualValues(t, []float32{3, 5, 7}, res.Vector)
+
+		helper.DeleteReference(t, article.Object(), ref1, "hasParagraphs")
+		res = helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.EqualValues(t, para2.Vector, res.Vector)
+
+		helper.DeleteReference(t, article.Object(), ref2, "hasParagraphs")
+		res = helper.AssertGetObject(t, article.Class, article.ID, "vector")
+		assert.Nil(t, res.Vector)
+	})
+}
+
+func newBeacon(className string, id strfmt.UUID) strfmt.URI {
+	return strfmt.URI(
+		fmt.Sprintf("weaviate://localhost/%s/%s",
+			className, id))
+}
```

Comment:
Please use the `crossref` package methods for managing beacons

Reply:
fixed